src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/SnowflakeQuery.scala

```diff
@@ -212,7 +212,9 @@ case class AggregateQuery(columns: Seq[NamedExpression],
       ConstantString("GROUP BY") +
         mkStatement(groups.map(expressionToStatement), ",")
     } else {
-      EmptySnowflakeSQLStatement()
+      // SNOW-201486: Insert a limit 1 to ensure that only one row returns if there
+      // are no grouping expressions
+      ConstantString("LIMIT 1").toStatement
```

Comment:
Do we ever generate having classes ? If so, this may lead to incorrect syntax.

Reply:
We do not generate `HAVING` clauses. These are always converted into `WHERE` clauses during planning.